collector/src/compile/execute/profiler.rs

```diff
@@ -51,6 +52,68 @@ impl Profiler {
                 | Profiler::DepGraph
         )
     }
+
+    pub fn diff_prereqs(&self) -> anyhow::Result<()> {
+        use Profiler::*;
+        match self {
+            Cachegrind => {
+                check_installed("valgrind")?;
+                check_installed("cg_annotate")?;
+            }
+
+            SelfProfile | PerfRecord | Oprofile | Samply | Callgrind | Dhat | DhatCopy | Massif
+            | Bytehound | Eprintln | LlvmLines | MonoItems | DepGraph | LlvmIr => {}
+        }
+        Ok(())
+    }
+
+    /// A file prefix added to all files of this profiler.
+    pub fn prefix(&self) -> &'static str {
+        use Profiler::*;
+        match self {
+            Cachegrind => "cgout",
+            DepGraph => "dep-graph",
+
+            SelfProfile | PerfRecord | Oprofile | Samply | Callgrind | Dhat | DhatCopy | Massif
+            | Bytehound | Eprintln | LlvmLines | MonoItems | LlvmIr => "",
+        }
+    }
+
+    /// A postfix added to the file that gets diffed.
+    pub fn postfix(&self) -> &'static str {
+        use Profiler::*;
+        match self {
+            Cachegrind => "",
+            DepGraph => ".txt",
+
+            SelfProfile | PerfRecord | Oprofile | Samply | Callgrind | Dhat | DhatCopy | Massif
+            | Bytehound | Eprintln | LlvmLines | MonoItems | LlvmIr => "",
+        }
+    }
+
+    /// A file prefix to use for the diff output.
+    pub fn output_prefix(&self) -> &'static str {
```

Comment:
Maybe `diff_prefix`?

But I think that we can also just unify `prefix` and `output_prefix` to make the code simpler, and always just append `-diff` to the prefix. We would need to change the rest of the code to fixup the cachegrind diff paths, but that's fine, I don't think that the specific `cgann-diff` name was that important.